### DISCUSSION TITLE
Next.js hydration error when used with emotion

### DISCUSSION THREAD

**Discussion by zaicevas** [system-ui/theme-ui#2088] [Q&A] [answered]
**Describe the bug**
`theme-ui` does not integrate with `emotion` and Next.JS. Getting hydration errors.

![image](https://user-images.githubusercontent.com/34719980/144748578-20fc3a71-a983-430f-afdc-2d4e7224b466.png)

**To Reproduce**

1. Bootstrap with https://github.com/lachlanjc/next-theme-starter
2. Integrate emotion according to https://github.com/vercel/next.js/tree/canary/examples/with-emotion
3. Run dev server

Reproducible repo: https://github.com/zaicevas/next-emotion-theme-ui

**Expected behavior**
No errors

Let me also point out that in this case there seems NOT to be any visual issues, but in another project I do get visual inconsistencies (light mode style is incorrectly used until manually toggled)

**Comment by lachlanjc** [comment, marked as answer]
Hey @zaicevas! You shouldn’t need to (afaik) update the configuration past what’s in my starter repo at all—server rendering is [automatic](https://emotion.sh/docs/ssr#nextjs) with the latest Emotion in Next. The `with-emotion` example setup will disable SWC & double-render styles, last I checked.
